src/coreclr/tools/aot/crossgen2/crossgen2_publish.csproj

```diff
@@ -2,12 +2,14 @@
   <!-- This project is publishes a self-contained copy of crossgen2. -->
   <PropertyGroup>
     <_IsPublishing>true</_IsPublishing>
-    <RuntimeIdentifier>$(PackageRID)</RuntimeIdentifier>
+    <RuntimeIdentifier Condition="'$(CrossBuild)' == 'true'">$(NETCoreSdkPortableRuntimeIdentifier)</RuntimeIdentifier>
+    <RuntimeIdentifier Condition="'$(CrossBuild)' != 'true'">$(PackageRID)</RuntimeIdentifier>
     <PublishDir>$(RuntimeBinDir)crossgen2-published/</PublishDir>
     <SelfContained>true</SelfContained>
     <PublishTrimmed>true</PublishTrimmed>
     <TrimmerSingleWarn>false</TrimmerSingleWarn>
     <AotOrSingleFile>true</AotOrSingleFile>
+    <_IsCrossHostPublish>true</_IsCrossHostPublish>
```

Comment:
@steveisok I think we are missing `'$(TargetsAndroid)' == 'true'` around configuring crossgen2 project build.